### ISSUE TITLE
TZ93 thermostat tweaks needed

### ISSUE THREAD

**Issue by Infiniverse** [pshotton/com.tkbhome#9]
Hi. Thanks for adding TZE93  support recently. I’ve managed to get one working now so it works.

A few quirks though.

* the image of the TZE93 shows the wrong device. Do you want a phone?
* when adding a the device the correct procedure is to press the left hand menu button three times - the instructions referred to a different button and had a different picture.
* I don’t know if this is a quirk or no, but the device widget in homey shows the value of the set point. Shouldn’t it show the current temperature? The current temperature is shown properly on the main device window.
* Finally, I can’t see how to switch the thermostat on and off from honey. This feature is possible the device, because I could do it within Fibaro Home Centre Controller.

If you need any help with any of this I would be grateful to look. I’m totally unfamiliar with the framework currently though, so am looking for support getting up to speed. Thanks.

**Comment by pshotton** [comment]
Hi, thanks for the useful feedback. I based the driver on the similar TKB
TZE96, but there are obvious differences. If you could send photos of the
device and the button pressing then I can update the driver info. Do you
have an instruction leaflet that you can scan?

I will look into the device widget - you should be able to choose what you
see but I'm not sure how that is programmed. Likewise switching on and off.

I'm not very familiar with the framework but I'm slowly learning! I'm a
Java consultant by trade and Javascript is not my core language, and the
documentation for Homey doesn't give detail on how to do things, but I
think the ecosystem is good and if I can add to it I will.

Thanks for the feedback
Phil

**Comment by Infiniverse** [comment]
[TZE93_manual_EN.pdf](https://github.com/pshotton/com.tkbhome/files/13697523/TZE93_manual_EN.pdf)

here’s the manual.

**Comment by Infiniverse** [comment]
Here’s the image.
![Uploading IMG_4565.jpeg…]()

**Comment by Infiniverse** [comment]
I’m a Java developer too. :)

**Comment by Infiniverse** [comment]
Regarding switching the thermostat on and off, it’s controlled with the mode command.

![IMG_4574](https://github.com/pshotton/com.tkbhome/assets/2773422/b8541261-dc59-4161-9c8e-ef3f6ae17ca8)

**Comment by Infiniverse** [comment]
Hey Phil. Merry Christmas mate. When you get a moment can you please let me know what your thoughts are on this, if any. If you’re too busy that’s cool too - but if you could let me know I’d be grateful. Cheers. Joe

**Comment by pshotton** [comment]
Hi Joe, and Merry Christmas to you.
I've just pushed a test version of the TZE93 driver to the Homey store,
I've added the "onoff" capability, which I think means that on-off basic
control should be added (I'm still trying to get my head around Homey
capabilities), and I've changed the images and pairing instructions to
match the TZ93 rather than the TZ96.

Can you give it a go and let me know if it works for you? If so I'll push
into release.

Many thanks
Phil

**Comment by Infiniverse** [comment]
Thanks Phil. You’re a star! (I’m not sure how to install a beta app! Can you please point me in the right direction?)

**Comment by pshotton** [comment]
Hi Joe

Install from this link
https://homey.app/en-us/app/com.tkbhome/TKB-Home/test/

I hope it works

All the best
Phil

**Comment by Infiniverse** [comment]
Thanks. That installed ok. I’m guessing I need to reregister the devices?

**Comment by Infiniverse** [comment]
Ok. I’ve added a new device. A few issues to report.

* the device type icon on the add device wizard is still the old one, but the image is right when the device is selected

* the device installs, but complains when I click on it: 
![IMG_4624](https://github.com/pshotton/com.tkbhome/assets/2773422/494c1564-dd6f-4344-935f-b2e5eca73106)

**Comment by pshotton** [comment]
Interesting...
Not sure what this means, I'll do some digging. I added the 'onoff'
capability but it looks like I missed something to do with handling it.
The documentation could be better!
Leave it with me...

**Comment by Infiniverse** [comment]
For reference, my TKBHome thermostat has a control panel that looks like this for setting he mode:

![IMG_4626](https://github.com/pshotton/com.tkbhome/assets/2773422/68bb27cc-6fac-4eec-8ceb-f3d9eba8f21d)

**Comment by Infiniverse** [comment]
On my Fibaro home centre this thermostat shows up as three devices. The heating mode has values Off/Heating/Resume. No idea what resume means.
![IMG_4627](https://github.com/pshotton/com.tkbhome/assets/2773422/e50508e3-273b-446c-a780-4fa2d1335ddb)


![IMG_4629](https://github.com/pshotton/com.tkbhome/assets/2773422/991d1162-95f5-463d-8e04-479ac7a868de)
![IMG_4628](https://github.com/pshotton/com.tkbhome/assets/2773422/3dc91616-5f3a-4ba1-a784-21d1ec1ee97f)

**Comment by pshotton** [comment]
I'm slowly getting my head around homey <-> Zwave command mapping...
I've just pushed a new version into test, could you give it a try and see
if it works?
Sorry for the trial and error approach but without a physical device it's
difficult to program!

**Comment by Infiniverse** [comment]
3.1.4 still has the missing on/off handler error. (I uninstalled and and reinstalled it just to be sure).

**Comment by pshotton** [comment]
OK, just pushed 3.1.5 into test. Another go at solving the onoff conundrum. I think you;ll need to uninstall and reinstall, but try without first if you like.

**Comment by Infiniverse** [comment]
Yes, that worked. I get events when I turn it on and off by hand, and can also change the mode from homey.

I also didn't need to repair any thermostats. They all support the mode change.

Thanks!

**Comment by Infiniverse** [comment]
(did you see my notes about the images during setup and the icon?

**Comment by pshotton** [comment]
Yes, I've submitted a request to Homey to generate the icons. They are
vector graphics and I have no idea how to create them but Homey has a
process for generating them.
I'll update once I've got them.

**Comment by Infiniverse** [comment]
As an aside. It looks like I do need to reregister my devices to get the new mode behaviour. The ones I setup before some point don’t have the mode switch option available.

**Comment by pshotton** [comment]
Hi. I've just released a new test version with updated icons and also a change to the onoff handler registration (I was getting errors when on state was changed). Can you please test and confirm it works properly. Thanks
_

**Comment by edsallo** [comment]
I have TZ10. 36 and TZE93.716 thermostats. They worked very well with Homey Pro (2019). Switching on/off also works.
Today I purchased a new Homey Pro (2023).
The application with which the thermostats worked on the new Homey is installed with errors and does not work.
I would be grateful if you add it to your application.
[2.0.0.zip](https://github.com/user-attachments/files/17451622/2.0.0.zip)

**Comment by edsallo** [comment]
![ IMG_1589](https://github.com/user-attachments/assets/b12c1667-94fe-42a9-94e4-c61461fa1b5c)

With the current application it looks like this.
